### PR TITLE
osd/ReplicatedPG: move classes to cc file

### PR DESCRIPTION
**Pull request by stiopaa1**
Move
C_OSD_OndiskWriteUnlock
C_OSD_AppliedRecoveredObject
C_OSD_CommittedPushedObject
C_OSD_AppliedRecoveredObjectReplica
to cc file.

Signed-off-by: Michal Jarzabek stiopa@gmail.com

### CODE REVIEW

**Comment by tchaikov** [comment]
lgtm once we have a green jenkins run.

File: src/osd/ReplicatedPG.h [hunk 1]
```diff
@@ -1271,50 +1271,10 @@ class ReplicatedPG : public PG, public PGBackend::Listener {
...
-  };
-  struct C_OSD_AppliedRecoveredObjectReplica : public Context {
-    ReplicatedPGRef pg;
-    explicit C_OSD_AppliedRecoveredObjectReplica(ReplicatedPG *p) :
-      pg(p) {}
-    void finish(int r) {
-      pg->_applied_recovered_object_replica();
-    }
-  };
+  struct C_OSD_OndiskWriteUnlock;
```
Comment (tchaikov) [inline src/osd/ReplicatedPG.h]: @stiopaa1  can we move these into the .cc file also? maybe we can even dispense with them, as they are used after they are defined.

**Comment by liewegas** [comment]
needs rebase

**Comment by dachary** [comment]
feel free to re-open once you have time to work on it

**Pull request closed without merging**